pkg/image/registryclient/client.go

```diff
@@ -34,7 +34,7 @@ import (
 type RepositoryRetriever interface {
 	// Repository returns a properly authenticated distribution.Repository for the given registry, repository
 	// name, and insecure toleration behavior.
-	Repository(ctx context.Context, registry *url.URL, repoName string, insecure bool) (distribution.Repository, error)
+	Repository(ctx context.Context, registry *url.URL, repoName string, insecure bool) (RepositoryWithLocation, error)
```

Comment:
Can you link to where this is required (use case in oc)?

Reply:
I went through all the possible places and I'm not seeing any particular cases, yet. I'll revert that bit back as well.

Reply:
this was requested by image-registry (@ricardomaraschini ) they use this

Reply:
https://github.com/openshift/library-go/pull/939#discussion_r613257953

Reply:
Image registry uses the `RepositoryRetriever` interface.

Reply:
Right, but from what I'm looking around this means keeping the interface untouched which is what I did in my last commit. We can always re-submit that change, if necessary in a followup.